### ISSUE TITLE
Do you train the Tesseract-OCR for GT&T symbols?

### ISSUE THREAD

**Issue by bin1guo** [LinYiHsin/recg_drawing#3]
Good job~  really meaningful for engineer drawing.
Tesseract-OCR you used to text recognition. I want to know do you do special training for symbols?  or only use the original model
I think the original Tesseract-OCR can't recognize the symbols. 
Do you have the training script? Thanks

**Comment by bin1guo** [comment]
Meanwhile,  Can it recognize both ENG and Chinese? how to generate eng_gdt.traindata? thanks

**Comment by LinYiHsin** [comment]
Hi,
I am using the built-in font files in Windows to train the 'traineddata' for Tesseract-OCR, and both AIGDT and AMGDT fonts include GD&T symbols. If you want to recognize Chinese characters, you can also use font files that contain Chinese characters for training. For the training method, you can refer to this article: https://vocus.cc/article/621de852fd89780001653bcd
Hope this helps you.

**Comment by bin1guo** [comment]
Thanks,  if the current model already covers AIGDT and AMGDT, I think I can reuse your model to recognize most GDT symbols.
For Chinese characters, I will try to train a new model. 
the view detection and annotation detection are based on Yolo v7, right?    I note that the computation time for annotation detection in Table3 is too long, do you think a simple model can work for view and annotation detection?  1-3s will be an acceptable time.  Thank you for your answers.

**Comment by LinYiHsin** [comment]
Indeed, it is based on YOLOv7. 
Regarding computation time, if the graphics are more complex or have more features, the processing time is indeed longer. Perhaps you can try using multi-thread or parallel computing during recognition to shorten the processing time. 
Hope this is helpful to you.

**Comment by bin1guo** [comment]
Can this model run on CPU platform?  Thanks

**Comment by LinYiHsin** [comment]
I've checked the YOLOv7 documentation, and it seems you can set the parameters in `detect.py`. To execute YOLO recognition using the CPU, you can add `--device cpu` (by default, it's `--device 0`). The relevant execution parts should be in `app\celery\tasks.py`, specifically on lines 84, 132, and 180. 
Hope this helps you.

**Comment by bin1guo** [comment]
> I've checked the YOLOv7 documentation, and it seems you can set the parameters in `detect.py`. To execute YOLO recognition using the CPU, you can add `--device cpu` (by default, it's `--device 0`). The relevant execution parts should be in `app\celery\tasks.py`, specifically on lines 84, 132, and 180. Hope this helps you.

Yeah, it works, thanks for your details!  Do you try this detect framework on more engineer drawing files? how about the generalization ability?

**Comment by bin1guo** [comment]
Another question needs your help,  I note that there is a Tess_OCR function in the file detect_fcf.py. and a separate file Tess_OCR.py. 
in Tess_OCR.py,  the rotation is done on the image. while, in detect_fcf.py, there is no rotation.   To detect symbols and characters in fcf,  should the Tess_OCR.py be used?  in my test, the view detection is good. but the dim detection and symbols rec is is good as shown in the paper. Thanks

**Comment by bin1guo** [comment]
![DIM_01](https://github.com/LinYiHsin/recg_drawing/assets/90182771/611b0eea-dc16-43b1-b756-d8860b338c9c)
![DIM_02](https://github.com/LinYiHsin/recg_drawing/assets/90182771/1c53d951-c106-4a0a-81c7-e0fffc9f8004)
![DIM_04](https://github.com/LinYiHsin/recg_drawing/assets/90182771/072a019e-bed7-4989-a6de-0e1f231a6426)
![DIM_05](https://github.com/LinYiHsin/recg_drawing/assets/90182771/6263e067-e727-4961-a92e-d49d377ce5af)
the Dim detect results

**Comment by LinYiHsin** [comment]
Regarding the issue with Tess_OCR.py, I feel that the annotation text for the Feature Control Frame and Datum Feature doesn't need to be rectified. That's why I haven't used Tess_OCR.py and focused solely on recognizing text. As for the effectiveness of recognizing annotations, it's possible that my training data differs significantly from the graphics you're trying to recognize. Perhaps you can try training with some data that better matches your requirements. 
Hope this helps you.

**Comment by bin1guo** [comment]
I see,  I saw there are training data in /yolov7/data folder.  Could you kindly provide the script for training? thanks

**Comment by LinYiHsin** [comment]
The training method for YOLOv7, please refer to this link: https://github.com/WongKinYiu/yolov7
